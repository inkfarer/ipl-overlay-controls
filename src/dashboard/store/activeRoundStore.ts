import { ActiveRound, SwapColorsInternally } from 'schemas';
import { createStore, Store, useStore } from 'vuex';
import cloneDeep from 'lodash/cloneDeep';
import { InjectionKey } from 'vue';
import { GameWinner } from 'types/enums/gameWinner';
import { SetActiveColorRequest, SetWinnerRequest, SwapRoundColorRequest } from 'types/messages/activeRound';
import { SetRoundRequest } from 'types/messages/rounds';
import { ActiveRoundGame } from 'types/activeRoundGame';

const activeRound = nodecg.Replicant<ActiveRound>('activeRound');
const swapColorsInternally = nodecg.Replicant<SwapColorsInternally>('swapColorsInternally');

export const activeRoundReps = [ activeRound, swapColorsInternally ];

export interface ActiveRoundStore {
    activeRound: ActiveRound;
    swapColorsInternally: SwapColorsInternally;
}

export const activeRoundStore = createStore<ActiveRoundStore>({
    state: {
        activeRound: null,
        swapColorsInternally: null
    },
    mutations: {
        setState(store, { name, val }: { name: string, val: unknown }): void {
            this.state[name] = cloneDeep(val);
        }
    },
    actions: {
        setWinner(store, { winner }: { winner: GameWinner }): void {
            nodecg.sendMessage('setWinner', { winner });
        },
        removeWinner(): void {
            nodecg.sendMessage('removeWinner');
        },
        setActiveColor(store, request: SetActiveColorRequest): void {
            nodecg.sendMessage('setActiveColor', request);
        },
        swapColors(): void {
            swapColorsInternally.value = !swapColorsInternally.value;
        },
        setWinnerForIndex(store, { index, winner }: { index: number, winner: GameWinner }): void {
            nodecg.sendMessage('setWinner', { winner, roundIndex: index } as SetWinnerRequest);
        },
        resetActiveRound(): void {
            nodecg.sendMessage('resetActiveRound');
        },
        setActiveRound(store, request: SetRoundRequest): void {
            nodecg.sendMessage('setActiveRound', request);
        },
        swapRoundColor(store, request: SwapRoundColorRequest): void {
            nodecg.sendMessage('swapRoundColor', request);
        },
        updateActiveGames(store, games: ActiveRoundGame[]): void {
            nodecg.sendMessage('updateActiveGames', { games });
        },
        switchToNextColor(): void {
            nodecg.sendMessage('switchToNextColor');
        },
        switchToPreviousColor(): void {
            nodecg.sendMessage('switchToPreviousColor');
        },
        async getNextAndPreviousColors(): Promise<void> {
            return nodecg.sendMessage('getNextAndPreviousColors');
        }
    }
});

export const activeRoundStoreKey: InjectionKey<Store<ActiveRoundStore>> = Symbol();

export function useActiveRoundStore(): Store<ActiveRoundStore> {
    return useStore(activeRoundStoreKey);
}
