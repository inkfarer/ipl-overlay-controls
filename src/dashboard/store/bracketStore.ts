import { Bracket } from '@tourneyview/common';
import { defineStore } from 'pinia';

const bracketData = nodecg.Replicant<Bracket | undefined>('bracketData');

export const bracketReps = [ bracketData ];

export interface BracketStore {
    bracketData: Bracket | undefined;
}

export const useBracketStore = defineStore('brackets', {
    state: () => ({
        bracketData: undefined
    } as BracketStore),
    actions: {
        resetBracketData() {
            bracketData.value = undefined;
        }
    }
});
