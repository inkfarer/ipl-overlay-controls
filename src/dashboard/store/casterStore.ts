import { NodeCGBrowser } from 'nodecg/browser';
import { Caster, Casters, RadiaSettings } from 'schemas';
import { generateId } from '../../helpers/generateId';
import { defineStore } from 'pinia';

const casters = nodecg.Replicant<Casters>('casters');
const radiaSettings = nodecg.Replicant<RadiaSettings>('radiaSettings');

export const castersReps = [ casters, radiaSettings ];

export interface CasterStore {
    casters: Casters;
    uncommittedCasters: Casters;
    radiaSettings: RadiaSettings;
}

const defaultCaster = {
    name: 'New Caster',
    twitter: '',
    pronouns: '?/?'
};

export const useCasterStore = defineStore('casters', {
    state: () => ({
        casters: {},
        uncommittedCasters: {},
        radiaSettings: {
            enabled: null,
            guildID: null,
            updateOnImport: null
        }
    } as CasterStore),
    actions: {
        updateCaster({ id, newValue }: { id: string, newValue: Caster }): void {
            casters.value[id] = newValue;
        },
        removeUncommittedCaster(id: string): void {
            delete this.uncommittedCasters[id];
        },
        addUncommittedCaster({ id, caster }: { id: string, caster?: Caster }): void {
            this.uncommittedCasters[id] = caster ?? defaultCaster;
        },
        addDefaultCaster(): string {
            const id = generateId();
            this.addUncommittedCaster({ id });
            return id;
        },
        async saveUncommittedCaster({ id, caster }: { id: string, caster: Caster }): Promise<string> {
            const newId = await nodecg.sendMessage('saveCaster', caster);
            this.removeUncommittedCaster(id);
            return newId;
        },
        async loadCastersFromVc(): Promise<void> {
            const result = await nodecg.sendMessage('getLiveCommentators');
            if (result.extra && result.extra.length > 0) {
                for (let i = 0; i < result.extra.length; i++) {
                    const extraCaster = result.extra[i];
                    const id = extraCaster.discord_user_id;
                    delete extraCaster.discord_user_id;
                    this.addUncommittedCaster({ id, caster: extraCaster });
                }
            }
        },
        showCasters() {
            nodecg.sendMessage('mainShowCasters');
        },
        async removeCaster(id: string): Promise<void> {
            return nodecg.sendMessage('removeCaster', { id });
        }
    }
});
