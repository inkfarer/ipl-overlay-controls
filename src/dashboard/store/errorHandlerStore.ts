import { App } from 'vue';
import { generateId } from '../../helpers/generateId';
import { defineStore } from 'pinia';

export const useErrorHandlerStore = defineStore('errorHandler', {
    state: () => ({
        recentErrors: {}
    }),
    actions: {
        removeRecentError({ key }: { key: string }): void {
            delete this.recentErrors[key];
        },
        handleError({ err, info }: { err: unknown, info: string }): void {
            console.error(`Got error from '${info}': \n`, err);

            if (Object.keys(this.recentErrors).length >= 2) return;

            const id = generateId();
            this.recentErrors[id] = err;
            window.setTimeout(() => {
                delete this.recentErrors[id];
            }, 25000);
        }
    }
});

export function setUpErrorHandler(app: App<unknown>): void {
    const store = useErrorHandlerStore();
    app.config.errorHandler = (err, vm, info) => {
        store.handleError({ err, info });
    };
}
