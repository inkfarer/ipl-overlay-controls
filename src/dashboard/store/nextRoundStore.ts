import { NextRound } from 'schemas';
import { SetNextRoundRequest } from 'types/messages/rounds';
import { BeginNextMatchRequest } from 'types/messages/activeRound';
import { defineStore } from 'pinia';

const nextRound = nodecg.Replicant<NextRound>('nextRound');

export const nextRoundReps = [ nextRound ];

export interface NextRoundStore {
    nextRound: NextRound;
}

export const useNextRoundStore = defineStore('nextRound', {
    state: () => ({
        nextRound: null
    } as NextRoundStore),
    actions: {
        setShowOnStream(newValue: boolean) {
            nextRound.value.showOnStream = newValue;
        },
        beginNextMatch(request: BeginNextMatchRequest): void {
            nodecg.sendMessage('beginNextMatch', request);
        },
        setNextRound(request: SetNextRoundRequest): void {
            nodecg.sendMessage('setNextRound', request);
        }
    }
});
