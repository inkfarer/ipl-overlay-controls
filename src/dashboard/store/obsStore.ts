import { NodeCGBrowser } from 'nodecg/browser';
import { ObsCredentials, ObsData } from 'schemas';
import { createStore, Store, useStore } from 'vuex';
import cloneDeep from 'lodash/cloneDeep';
import { InjectionKey } from 'vue';
import { SetObsDataRequest } from 'types/messages/obs';

const obsData = nodecg.Replicant<ObsData>('obsData');
const obsCredentials = nodecg.Replicant<ObsCredentials>('obsCredentials');

export const obsReps = [obsData, obsCredentials];

export interface ObsStore {
    obsData: ObsData
    obsCredentials: ObsCredentials
}

export const obsStore = createStore<ObsStore>({
    state: {
        obsData: null,
        obsCredentials: null
    },
    mutations: {
        setState(store, { name, val }: { name: string, val: unknown }): void {
            this.state[name] = cloneDeep(val);
        }
    },
    actions: {
        async connect(store, { address, password }: { address: string, password?: string }): Promise<void> {
            return nodecg.sendMessage('connectToObs', { address, password });
        },
        async setData(store, data: SetObsDataRequest): Promise<void> {
            return nodecg.sendMessage('setObsData', data);
        },
        async startGame(): Promise<void> {
            return nodecg.sendMessage('startGame');
        },
        async endGame(): Promise<void> {
            return nodecg.sendMessage('endGame');
        },
        setEnabled(store, enabled: boolean): void {
            nodecg.sendMessage('setObsSocketEnabled', enabled);
        }
    }
});

export const obsStoreKey: InjectionKey<Store<ObsStore>> = Symbol();

export function useObsStore(): Store<ObsStore> {
    return useStore(obsStoreKey);
}
