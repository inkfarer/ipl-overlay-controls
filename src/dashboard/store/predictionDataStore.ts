import { NodeCGBrowser } from 'nodecg/browser';
import { Prediction, PredictionStore } from 'schemas';
import { PredictionStatus } from 'types/enums/predictionStatus';
import { defineStore } from 'pinia';

const predictionStore = nodecg.Replicant<PredictionStore>('predictionStore');

export const predictionReps = [ predictionStore ];

export interface PredictionDataStore {
    predictionStore: PredictionStore;
}

export const usePredictionDataStore = defineStore('prediction', {
    state: () => ({
        predictionStore: null,
    } as PredictionDataStore),
    actions: {
        async lockPrediction() {
            return patchPrediction(
                this.predictionStore.currentPrediction,
                PredictionStatus.LOCKED,
                [ PredictionStatus.ACTIVE ]);
        },
        async cancelPrediction() {
            return patchPrediction(
                this.predictionStore.currentPrediction,
                PredictionStatus.CANCELED,
                [ PredictionStatus.ACTIVE, PredictionStatus.LOCKED ]);
        },
        async resolvePrediction({ winningOutcomeIndex }: { winningOutcomeIndex: number }) {
            if (winningOutcomeIndex > 1 || winningOutcomeIndex < 0) {
                throw new Error(`Cannot resolve prediction with outcome index ${winningOutcomeIndex}.`);
            } else if (!this.predictionStore.currentPrediction?.id) {
                throw new Error('No prediction to resolve.');
            } else if (this.predictionStore.currentPrediction.status !== PredictionStatus.LOCKED ) {
                throw new Error('Can only resolve a locked prediction.');
            }

            return nodecg.sendMessage('patchPrediction', {
                id: this.predictionStore.currentPrediction.id,
                status: PredictionStatus.RESOLVED,
                winning_outcome_id: this.predictionStore.currentPrediction.outcomes[winningOutcomeIndex].id
            });
        },
        async createPrediction(
            { title, teamAName, teamBName, duration }:
                { title: string, teamAName: string, teamBName: string, duration: number }
        ) {
            if (
                [PredictionStatus.ACTIVE, PredictionStatus.LOCKED]
                    .includes(this.predictionStore.currentPrediction?.status as PredictionStatus)
            ) {
                throw new Error('An unresolved prediction already exists.');
            }

            return nodecg.sendMessage('postPrediction', {
                title,
                outcomes: [
                    { title: teamAName },
                    { title: teamBName }
                ],
                prediction_window: duration
            });
        },
        async reconnect() {
            return nodecg.sendMessage('reconnectToRadiaSocket');
        }
    }
});

async function patchPrediction(
    currentPrediction: Prediction,
    status: PredictionStatus,
    allowedForStatuses: PredictionStatus[]
): Promise<void> {
    if (!currentPrediction || !allowedForStatuses.includes(currentPrediction?.status as PredictionStatus)) {
        throw new Error(`Cannot set prediction status at this time. Status must match one of [${allowedForStatuses.join(', ')}]`);
    }

    return nodecg.sendMessage('patchPrediction', {
        id: currentPrediction.id,
        status: status,
    });
}
