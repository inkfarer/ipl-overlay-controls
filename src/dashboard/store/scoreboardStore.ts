import { NodeCGBrowser } from 'nodecg/browser';
import { ScoreboardData } from 'schemas';
import { defineStore } from 'pinia';

const scoreboardData = nodecg.Replicant<ScoreboardData>('scoreboardData');

export const scoreboardReps = [scoreboardData];

export interface ScoreboardStore {
    scoreboardData: ScoreboardData
}

export const useScoreboardStore = defineStore('scoreboard', {
    state: () => ({
        scoreboardData: {
            flavorText: '',
            isVisible: null
        }
    } as ScoreboardStore),
    actions: {
        setFlavorText(flavorText: string): void {
            scoreboardData.value.flavorText = flavorText;
        },
        setScoreboardVisible(isVisible: boolean): void {
            scoreboardData.value.isVisible = isVisible;
        }
    }
});
