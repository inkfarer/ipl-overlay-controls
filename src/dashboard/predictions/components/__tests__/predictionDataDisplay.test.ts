import PredictionDataDisplay from '../predictionDataDisplay.vue';
import { usePredictionDataStore } from '../../../store/predictionDataStore';
import { config, mount } from '@vue/test-utils';
import { PredictionStatus } from 'types/enums/predictionStatus';
import { DateTime } from 'luxon';
import { createTestingPinia, TestingPinia } from '@pinia/testing';

describe('PredictionDataDisplay', () => {
    let pinia: TestingPinia;

    config.global.stubs = {
        FontAwesomeIcon: true,
        IplProgressBar: true
    };

    beforeEach(() => {
        pinia = createTestingPinia();

        usePredictionDataStore().$state = {
            predictionStore: {
                status: {
                    socketOpen: true,
                    predictionsEnabled: true
                }
            }
        };
    });

    it('matches snapshot', () => {
        const store = usePredictionDataStore();
        store.predictionStore.currentPrediction = {
            id: 'prediction123',
            broadcasterId: 'ipl',
            broadcasterName: 'IPL',
            broadcasterLogin: 'eye pee el',
            title: 'Who will win?',
            outcomes: [
                {
                    id: 'outcome-1',
                    title: 'First Team',
                    users: 5,
                    pointsUsed: 10000,
                    topPredictors: [],
                    color: 'BLUE'
                },
                {
                    id: 'outcome-2',
                    title: 'Second Team',
                    users: 1,
                    pointsUsed: 1,
                    topPredictors: [],
                    color: 'PINK'
                }
            ],
            duration: 60,
            status: PredictionStatus.ACTIVE,
            creationTime: '2020'
        };
        const wrapper = mount(PredictionDataDisplay, {
            global: {
                plugins: [pinia]
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('matches snapshot with newly started prediction', () => {
        const store = usePredictionDataStore();
        store.predictionStore.currentPrediction = {
            id: 'prediction123',
            broadcasterId: 'ipl',
            broadcasterName: 'IPL',
            broadcasterLogin: 'eye pee el',
            title: 'Who will win?',
            outcomes: [
                {
                    id: 'outcome-1',
                    title: 'First Team',
                    users: 5,
                    pointsUsed: null,
                    topPredictors: [],
                    color: 'BLUE'
                },
                {
                    id: 'outcome-2',
                    title: 'Second Team',
                    users: 1,
                    pointsUsed: null,
                    topPredictors: [],
                    color: 'PINK'
                }
            ],
            duration: 60,
            status: PredictionStatus.ACTIVE,
            creationTime: '2020'
        };
        const wrapper = mount(PredictionDataDisplay, {
            global: {
                plugins: [pinia]
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('matches snapshot with winner', () => {
        const store = usePredictionDataStore();
        store.predictionStore.currentPrediction = {
            id: 'prediction123',
            broadcasterId: 'ipl',
            broadcasterName: 'IPL',
            broadcasterLogin: 'eye pee el',
            title: 'Who will win?',
            winningOutcome: 'outcome-2',
            outcomes: [
                {
                    id: 'outcome-1',
                    title: 'First Team',
                    users: 5,
                    pointsUsed: 10000,
                    topPredictors: [],
                    color: 'BLUE'
                },
                {
                    id: 'outcome-2',
                    title: 'Second Team',
                    users: 20,
                    pointsUsed: 45879,
                    topPredictors: [],
                    color: 'PINK'
                }
            ],
            duration: 60,
            status: PredictionStatus.ACTIVE,
            creationTime: '2020'
        };
        const wrapper = mount(PredictionDataDisplay, {
            global: {
                plugins: [pinia]
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('matches snapshot with timer', () => {
        jest.spyOn(window, 'setInterval').mockImplementation((callback) => {
            (callback as () => void)();
            // In the browser, setInterval returns a number.
            return 0;
        });
        const store = usePredictionDataStore();
        store.predictionStore.currentPrediction = {
            id: 'prediction123',
            broadcasterId: 'ipl',
            broadcasterName: 'IPL',
            broadcasterLogin: 'eye pee el',
            title: 'Who will win?',
            outcomes: [
                {
                    id: 'outcome-1',
                    title: 'First Team',
                    users: 5,
                    pointsUsed: 10000,
                    topPredictors: [],
                    color: 'BLUE'
                },
                {
                    id: 'outcome-2',
                    title: 'Second Team',
                    users: 1,
                    pointsUsed: 1,
                    topPredictors: [],
                    color: 'PINK'
                }
            ],
            duration: 60,
            status: PredictionStatus.ACTIVE,
            creationTime: '2020',
            lockTime: DateTime.now().plus({ seconds: 100 }).toISO()
        };
        const wrapper = mount(PredictionDataDisplay, {
            global: {
                plugins: [pinia]
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });
});
