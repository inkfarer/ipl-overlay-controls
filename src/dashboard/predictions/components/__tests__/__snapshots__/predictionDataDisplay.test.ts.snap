// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PredictionDataDisplay matches snapshot 1`] = `
"<div data-v-d255ffee="" class="ipl-space color-dark">
  <div class="title">Who will win?</div>
  <!--v-if-->
  <div data-v-d255ffee="" class="ipl-space color-light prediction-outcomes-space">
    <div class="outcome"><span class="team-name wrap-anywhere">First Team</span>
      <div><span>10,000 points</span><span style="float: right;">5 <font-awesome-icon-stub icon="users" border="false" fixedwidth="false" flip="false" listitem="false" pulse="false" swapopacity="false" spin="false" symbol="false" inverse="false" bounce="false" shake="false" beat="false" fade="false" beatfade="false" flash="false" spinpulse="false" spinreverse="false"></font-awesome-icon-stub> 100% </span></div>
      <ipl-progress-bar-stub color="blue" backgroundcolor="dark" class="m-t-4" value="100"></ipl-progress-bar-stub>
    </div>
    <div class="outcome">
      <ipl-progress-bar-stub color="pink" backgroundcolor="dark" value="0"></ipl-progress-bar-stub>
      <div class="m-t-4"><span>1 point</span><span style="float: right;">1 <font-awesome-icon-stub icon="users" border="false" fixedwidth="false" flip="false" listitem="false" pulse="false" swapopacity="false" spin="false" symbol="false" inverse="false" bounce="false" shake="false" beat="false" fade="false" beatfade="false" flash="false" spinpulse="false" spinreverse="false"></font-awesome-icon-stub> 0% </span></div><span class="team-name wrap-anywhere">Second Team</span>
    </div>
  </div>
</div>"
`;

exports[`PredictionDataDisplay matches snapshot with newly started prediction 1`] = `
"<div data-v-d255ffee="" class="ipl-space color-dark">
  <div class="title">Who will win?</div>
  <!--v-if-->
  <div data-v-d255ffee="" class="ipl-space color-light prediction-outcomes-space">
    <div class="outcome"><span class="team-name wrap-anywhere">First Team</span>
      <div><span>0 points</span><span style="float: right;">5 <font-awesome-icon-stub icon="users" border="false" fixedwidth="false" flip="false" listitem="false" pulse="false" swapopacity="false" spin="false" symbol="false" inverse="false" bounce="false" shake="false" beat="false" fade="false" beatfade="false" flash="false" spinpulse="false" spinreverse="false"></font-awesome-icon-stub> 0% </span></div>
      <ipl-progress-bar-stub color="blue" backgroundcolor="dark" class="m-t-4" value="0"></ipl-progress-bar-stub>
    </div>
    <div class="outcome">
      <ipl-progress-bar-stub color="pink" backgroundcolor="dark" value="0"></ipl-progress-bar-stub>
      <div class="m-t-4"><span>0 points</span><span style="float: right;">1 <font-awesome-icon-stub icon="users" border="false" fixedwidth="false" flip="false" listitem="false" pulse="false" swapopacity="false" spin="false" symbol="false" inverse="false" bounce="false" shake="false" beat="false" fade="false" beatfade="false" flash="false" spinpulse="false" spinreverse="false"></font-awesome-icon-stub> 0% </span></div><span class="team-name wrap-anywhere">Second Team</span>
    </div>
  </div>
</div>"
`;

exports[`PredictionDataDisplay matches snapshot with timer 1`] = `
"<div data-v-d255ffee="" class="ipl-space color-dark">
  <div class="title">Who will win?</div>
  <div class="text-center">1:39 remaining </div>
  <div data-v-d255ffee="" class="ipl-space color-light prediction-outcomes-space">
    <div class="outcome"><span class="team-name wrap-anywhere">First Team</span>
      <div><span>10,000 points</span><span style="float: right;">5 <font-awesome-icon-stub icon="users" border="false" fixedwidth="false" flip="false" listitem="false" pulse="false" swapopacity="false" spin="false" symbol="false" inverse="false" bounce="false" shake="false" beat="false" fade="false" beatfade="false" flash="false" spinpulse="false" spinreverse="false"></font-awesome-icon-stub> 100% </span></div>
      <ipl-progress-bar-stub color="blue" backgroundcolor="dark" class="m-t-4" value="100"></ipl-progress-bar-stub>
    </div>
    <div class="outcome">
      <ipl-progress-bar-stub color="pink" backgroundcolor="dark" value="0"></ipl-progress-bar-stub>
      <div class="m-t-4"><span>1 point</span><span style="float: right;">1 <font-awesome-icon-stub icon="users" border="false" fixedwidth="false" flip="false" listitem="false" pulse="false" swapopacity="false" spin="false" symbol="false" inverse="false" bounce="false" shake="false" beat="false" fade="false" beatfade="false" flash="false" spinpulse="false" spinreverse="false"></font-awesome-icon-stub> 0% </span></div><span class="team-name wrap-anywhere">Second Team</span>
    </div>
  </div>
</div>"
`;

exports[`PredictionDataDisplay matches snapshot with winner 1`] = `
"<div data-v-d255ffee="" class="ipl-space color-dark">
  <div class="title">Who will win?</div>
  <!--v-if-->
  <div data-v-d255ffee="" class="ipl-space color-light prediction-outcomes-space">
    <div class="outcome"><span class="team-name wrap-anywhere">First Team</span>
      <div><span>10,000 points</span><span style="float: right;">5 <font-awesome-icon-stub icon="users" border="false" fixedwidth="false" flip="false" listitem="false" pulse="false" swapopacity="false" spin="false" symbol="false" inverse="false" bounce="false" shake="false" beat="false" fade="false" beatfade="false" flash="false" spinpulse="false" spinreverse="false"></font-awesome-icon-stub> 18% </span></div>
      <ipl-progress-bar-stub color="blue" backgroundcolor="dark" class="m-t-4" value="18"></ipl-progress-bar-stub>
    </div>
    <div class="outcome is-winner">
      <ipl-progress-bar-stub color="yellow" backgroundcolor="dark" value="82"></ipl-progress-bar-stub>
      <div class="m-t-4"><span>45,879 points</span><span style="float: right;">20 <font-awesome-icon-stub icon="users" border="false" fixedwidth="false" flip="false" listitem="false" pulse="false" swapopacity="false" spin="false" symbol="false" inverse="false" bounce="false" shake="false" beat="false" fade="false" beatfade="false" flash="false" spinpulse="false" spinreverse="false"></font-awesome-icon-stub> 82% </span></div><span class="team-name wrap-anywhere">Second Team</span>
    </div>
  </div>
</div>"
`;
